lib/excon/ssl_socket.rb

```diff
@@ -66,16 +66,20 @@ def initialize(data = {})
       end
 
       # maintain existing API
-      certificate_path = @data[:client_cert] || @data[:certificate_path]
-      private_key_path = @data[:client_key] || @data[:private_key_path]
+      certificate_data = @data[:client_cert] || @data[:certificate_path]
```

Comment:
How would you feel about just making a new more explicit key here? ie it might look like:
`certificate_data = @data[:client_cert_data] || @data[:client_cert] && File.read(@data[:client_cert]) || @data[:certificate_path]`.

I think on some level I prefer this (you explicitly opt in to path-or-data, instead of us needing to implicitly figure it out based on contents of the string.

What do you think?